### ISSUE TITLE
Plotly backend seems to fail

### ISSUE THREAD

**Issue by zerothi** [dtudk/dtumathtools#12]
See in the workflows: 
```
self = <spb.backends.plotly.plotly.PlotlyBackend object at 0x7efe82b34950>
series = (<spb.series.List2DSeries object at 0x7efe82b35e10>,)
kwargs = {'is_point': True, 'show': False}

    def __init__(self, *series, **kwargs):
        self.np = import_module('numpy')
        self.plotly = import_module(
            'plotly',
            import_kwargs={'fromlist': ['graph_objects', 'figure_factory']},
            warn_not_installed=True,
            min_module_version='5.0.0')
>       self.go = self.plotly.graph_objects
E       AttributeError: 'NoneType' object has no attribute 'graph_objects'

/opt/hostedtoolcache/Python/3.11.4/x64/lib/python3.11/site-packages/spb/backends/plotly/plotly.py:188: AttributeError
```
The CI does not list plotly as being installed, which makes me wonder whether there is a dependency mismatch.

**Comment by Chrillebon** [comment]
The change to `pyproject.toml` changed the import. The previous version with `setup.py` and `test.yml` installed the package using the tag [dev], which also included the plotly backend along with all the others. If this is to be replicated in the `pyproject.toml`, we need to have the install in the `test.yaml` also install all packages (and dev requirements).

Additionally: please, please, PLEASE, **always use branches** and then merging, such that the code in main ALWAYS WORKS!

**Comment by zerothi** [comment]
Yeah, I know. But we are too much in a hurry ... :(
Also, I have had bad experiences with branches and the workflows mixing things up.

Things work now so it shouldn't be a problem.

The point of the tests should be that they test what the user has requested, and testing for everything is not appropriate in this case. Either all the `extras` should be removed so there is only one extra that installs everything, so here the tests are not compatible with how the package works. I.e. nobody can do `pip install dtumathtools` and run tests successfully. That is my main concern.

**Comment by Chrillebon** [comment]
In that case, one backend has to be installed that works for all computers for both 2D and 3D. Last year, the proposed backend was PyQt5, but that has issues where [some computers cannot download the package](https://github.com/dtudk/dtumathtools/issues/4). Because of these issues, the main package does not include any backends but will have to be installed manually (or using the `extras`) later. If (for instance) plotly works *for all computers* instead of PyQt5, this would be the ideal way to go forward. In this case, it should also be aligned with the demos used in the course.

**Comment by zerothi** [comment]
Agreed, but how was this accepted in the course? Was it problematic, or? Did students try a lot of different variations before it worked, etc.?

**Comment by Chrillebon** [comment]
Students are still able to run all commands and all plots using no backends. However, as they will not be able to have the plots "pop out" and rotate them to get 3D information, it is not _optimal_. However, in the current state, the "standard download" is a _minimum viable product_. Some students found ways to work around and still install the PyQt5 package, whilst others still found the plotly backend to be optimal for 3D plotting, and simply changed to it. From my understanding, we will be moving forward with **plotly** as the default 3D backend, but this is yet to be confirmed and tested before being rolled out.

**Comment by zerothi** [comment]
ok, that is a relief.
